client/components/data/cart/index.jsx

```diff
@@ -17,7 +21,30 @@ function getStateFromStores() {
 	};
 }
 
-var CartData = React.createClass( {
+const CartData = React.createClass( {
+	componentWillMount: function() {
+		this.updateSidebarVisibility();
+	},
+
+	componentDidMount: function() {
+		CartStore.on( 'change', this.updateSidebarVisibility );
+	},
+
+	componentWillUnmount: function() {
+		CartStore.off( 'change', this.updateSidebarVisibility );
+	},
+
+	updateSidebarVisibility: function() {
+		const cart = CartStore.get(),
+			oneProductToCheckout = cart.hasLoadedFromServer && cart.products.length === 1;
+
+		if ( ! this.props.isCheckout || abtest( 'sidebarOnCheckoutOfOneProduct' ) === 'original' ) {
+			return;
+		}
+
+		this.props.setSection( null, { hasSidebar: ! oneProductToCheckout } );
```

Comment:
Is there a reason that this logic can't live in `Checkout` (probably in `componentDidMount` and `componentWillReceiveProps`)?

- It feels slightly strange to wrap `CartData`, a component that already wraps `StoreConnection`, with `connect`.
- We wouldn't need the `isCheckout` prop.
- We wouldn't need to worry about setting/removing event listeners on `CartStore`, because `Checkout` receives new props every time cart data changes.